web/client/reducers/config.js

```diff
@@ -21,7 +21,12 @@ function mapConfig(state = null, action) {
         let hasVersion = action.config.version && action.config.version >= 2;
             // we get from the configuration what will be used as the initial state
         let mapState = action.legacy && !hasVersion ? ConfigUtils.convertFromLegacy(action.config) : ConfigUtils.normalizeConfig(action.config.map);
-
+        mapState.layers = mapState.layers.map( l => {
+            if (l.group === "background" && (l.type === "ol" || l.type === "OpenLayers.Layer")) {
+                l.type = "empty";
```

Comment:
You are mutating an object.
Moreover I would not do the conversion in a reducer, but here:  https://github.com/geosolutions-it/MapStore2/blob/master/web/client/utils/ConfigUtils.js#L151

Reply:
the problem is that new maps have these old values as type for empty background. so checking in the legacy converter only is not sufficient.  i'll provide a change to avoid mutate the object using assign()